src/DictVectors/pdvec.jl

```diff
@@ -0,0 +1,687 @@
+
+### Example
+
+```julia
+julia> add = FermiFS2C((1,1,0,0), (0,0,1,1));
+
+julia> op = HubbardMom1D(add; t=4/π^2, u=4);
+
+julia> pv = PDVec(add => 1.0)
+1-element PDVec: style = IsDeterministic{Float64}()
+  fs"|↑↑↓↓⟩" => 1.0
+
+julia> pv = op * pv
+7-element PDVec: style = IsDeterministic{Float64}()
+  fs"|↑↓↑↓⟩" => 1.0
+  fs"|↑↑↓↓⟩" => 4.0
+  fs"|↓↑↓↑⟩" => 1.0
+  fs"|↓↑↑↓⟩" => -1.0
+  fs"|⇅⋅⋅⇅⟩" => 1.0
+  fs"|↑↓↓↑⟩" => -1.0
+  fs"|⋅⇅⇅⋅⟩" => 1.0
+
+julia> map!(x -> -x, values(pv)); pv
+7-element PDVec: style = IsDeterministic{Float64}()
+  fs"|↑↓↑↓⟩" => -1.0
+  fs"|↑↑↓↓⟩" => -4.0
+  fs"|↓↑↓↑⟩" => -1.0
+  fs"|↓↑↑↓⟩" => 1.0
+  fs"|⇅⋅⋅⇅⟩" => -1.0
+  fs"|↑↓↓↑⟩" => 1.0
+  fs"|⋅⇅⇅⋅⟩" => -1.0
+
+julia> dest = similar(pv)
+0-element PDVec: style = IsDeterministic{Float64}()
+
+julia> map!(x -> x + 2, dest, values(pv))
+7-element PDVec: style = IsDeterministic{Float64}()
+  fs"|↑↓↑↓⟩" => 1.0
+  fs"|↑↑↓↓⟩" => -2.0
+  fs"|↓↑↓↑⟩" => 1.0
+  fs"|↓↑↑↓⟩" => 3.0
+  fs"|⇅⋅⋅⇅⟩" => 1.0
+  fs"|↑↓↓↑⟩" => 3.0
+  fs"|⋅⇅⇅⋅⟩" => 1.0
+
+julia> sum(values(pv))
+-6.0
+
+julia> dot(dest, pv)
+10.0
+
+julia> dot(dest, op, pv)
+44.0
+```
+
+## MPI
+
+When MPI is active, all parallel reductions are automatically reduced across MPI ranks
+with a call to `MPI.Allreduce`.
+
+In a distributed setting, `PDVec` does not support iteration without first making it
+explicit the iteration is only to be performed on the local segments of the vector. This is
+done with [`localpart`](@ref). In general, even when not using MPI, it is best practice to
+use [`localpart`](@ref) when explicit iteration is required.
+
+## Use with KrylovKit
+
+`PDVec` can be used with [KrylovKit.jl](https://github.com/Jutho/KrylovKit.jl) to find
+the eigenpairs of an operator in a matrix-free manner. For best performance, wrap the
+operator in `DictVectors.OperatorMulPropagator` first (see example below).
+
+Performing the `eigsolve` this way allow it to be run in a threaded and distributed manner.
+Using multiple MPI ranks with this method does not distribute the memory load effectively,
+but does result in significant speedups.
+
+### Example
+
+```julia
+julia> using KrylovKit
+
+julia> add = BoseFS((0,0,5,0,0));
+
+julia> op = HubbardMom1D(add; u=6.0);
+
+julia> pv = PDVec(add => 1.0);
+
+julia> propagator = DictVectors.OperatorMulPropagator(op, pv);
+
+julia> results = eigsolve(propagator, pv, 4, :SR; issymmetric=true);
+
+julia> results[1][1:4]
+4-element Vector{Float64}:
+ -3.4311156892322234
+  1.1821748602612363
+  3.7377753753082823
+  6.996390417443125
+```
+
+## Parallel functionality
+
+The following functions are parallelised and MPI-compatible:
+
+* [`mapreduce`](@ref) and derivatives (`sum`, `prod`, `reduce`...),
+* [`all`](@ref), [`any`](@ref),
+* [`map!`](@ref) (on values only),
+* `rmul!`, `lmul!`, `mul!`, `*`,
+* `add!`, `axpy!`, `axpby!`, `+`, `-`,
+* `dot`,
+* `*`, [`mul!`](@ref) and [`dot!`](@ref) with operators.
```

Comment:
In the documenter-generated documentation web page all the links in this part are broken. Unless there is a universal way to link to the standard Julia documentation, maybe it is best to not link them?
```suggestion
* `mapreduce` and derivatives (`sum`, `prod`, `reduce`...),
* `all`, `any`,
* `map!` (on values only),
*  `+`, `-`, `*` with scalars and operators
From the `LinearAlgebra` standard library:
* `rmul!`, `lmul!`, `mul!`, 
* `add!`, `axpy!`, `axpby!`, 
* `dot` with `PDVec`s and with operators.
```

Also, it would be good to indicate which of these functions are `Base` or always-available Julia function and which others belong to the `LinearAlgebra` standard library. Otherwise a reader who does not know about `axpby!` might get confused.